utils.py

```diff
@@ -38,3 +38,12 @@ def load_model(device, model_path, use_half):
     if use_half:
         model = model.half()
     return model
+
+
+def remove_parallel_wrapper(model):
+    """
+    Return the model or extract the model out of the parallel wrapper
+    :param model: The training model
+    :return: The model without parallel wrapper
+    """
+    return model.module if isinstance(model, torch.nn.parallel.DistributedDataParallel) else model
```

Comment:
`torch.nn.parallel.DistributedDataParallel` MUST be replaced with ` apex.parallel.DistributedDataParallel` and this is what happened when I executed the following print statements inside `train.py`:

```python
print('isinstance(model, torch.nn.parallel.DistributedDataParallel): ', isinstance(model, torch.nn.parallel.DistributedDataParallel))
print('isinstance(model, apex.parallel.DistributedDataParallel): ', isinstance(model, apex.parallel.DistributedDataParallel))
```

The output that I got is: 
```bash
isinstance(model, torch.nn.parallel.DistributedDataParallel):  False
isinstance(model, apex.parallel.DistributedDataParallel):  True
```